custom_components/deebot/hub.py

```diff
@@ -0,0 +1,77 @@
+import asyncio
+import random
+import logging
+import threading
+import async_timeout
+import string
+from datetime import timedelta
+from homeassistant.util import Throttle
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from deebotozmo import EcoVacsAPI, VacBot
+from .const import *
+
+_LOGGER = logging.getLogger(__name__)
+
+# Generate a random device ID on each bootup
+DEEBOT_API_DEVICEID = "".join(
+    random.choice(string.ascii_uppercase + string.digits) for _ in range(8)
+)
+
+
+class DeebotHub:
+    """Deebot Hub"""
+
+    def __init__(self, hass, domain_config):
+        """Initialize the Deebot Vacuum."""
+
+        self.config = domain_config
+        self._lock = threading.Lock()
+        self.hass = hass
+
+        self.ecovacs_api = EcoVacsAPI(
+            DEEBOT_API_DEVICEID,
+            domain_config.get(CONF_USERNAME),
+            EcoVacsAPI.md5(domain_config.get(CONF_PASSWORD)),
+            domain_config.get(CONF_COUNTRY),
+            domain_config.get(CONF_CONTINENT),
+        )
+
+        devices = self.ecovacs_api.devices()
+
+        liveMapEnabled = domain_config.get(CONF_LIVEMAP)
+        self.liveMapEnabled = liveMapEnabled
+
+        liveMapRooms = domain_config.get(CONF_SHOWCOLORROOMS)
+        country = domain_config.get(CONF_COUNTRY).lower()
+        continent = domain_config.get(CONF_CONTINENT).lower()
+        self.vacbots = []
+
+        # CREATE VACBOT FOR EACH DEVICE
+        for device in devices:
+            if device["name"] in domain_config.get(CONF_DEVICEID)[CONF_DEVICEID]:
+                vacbot = VacBot(
+                    self.ecovacs_api.uid,
+                    self.ecovacs_api.resource,
+                    self.ecovacs_api.user_access_token,
+                    device,
+                    country,
+                    continent,
+                    liveMapEnabled,
+                    liveMapRooms,
+                )
+
+                _LOGGER.debug("New vacbot found: " + device["name"])
+                vacbot.setScheduleUpdates()
```

Comment:
We should run this method later in a different thread. This method takes more then 10 seconds to complete.... Think you have more than one robot it can take more then 30 seconds or even more.
I think we should run this method later on in parallel. For testing I comment out this method and the conflig flow was quickly finished. Also I think the user has no problems if the robots is unavailable for a minutes after after creation.

Also `setScheduleUpdates` has problems, when the vacuum is offline during setup.

Reply:
I would probably remove this method. See https://github.com/And3rsL/Deebotozmo/pull/27#discussion_r592351927